src/libraries/System.Private.CoreLib/src/System/Globalization/CultureInfo.cs

```diff
@@ -1052,7 +1052,8 @@ public static CultureInfo GetCultureInfo(string name)
 
             lock (nameTable)
             {
-                nameTable[name] = result;
+                // add only if it wasn't already added
+                nameTable.TryAdd(name, result);
```

Comment:
Is there any chance the result would be different here for the case that it already exists and if so would that be observable based on ordering?

Reply:
It shouldn't, as we check the cache before creating a new entry. `und` is a special case we started supporting in .NET 10, and it happens to normalize to an empty string since there's no actual culture named `und`. And it will be wrong to overwrite the cached entry.

Reply:
I see - I was just wondering if `und` was the first and that happened to get added to the cache and was different in some observable way.

Reply:
> I was just wondering if und was the first and that happened to get added to the cache and was different in some observable way.

With this fix, `und` will be created exactly like the invariant culture. So, if `und` gets cached first, it's not an issue since it will hold the same invariant values. You can now think of `und` as an alias for the invariant culture.